### PR TITLE
Features/my mini bank

### PR DESCRIPTION
**Pull request by john-afiukwa**
done with the layouts

### CODE REVIEW

File: mini-bank/create-account.html [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<!DOCTYPE html>
```
Comment (nwankwo-ikemefuna) [inline mini-bank/create-account.html]: This is the landing page when the user has no account. The file name should reflect its function. So rename to `landing.html`

File: mini-bank/create-account.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    <link rel="stylesheet" href="../helpers/css/components.css">
+    <link rel="stylesheet" href="mini-bank.css">
+</head>
+<body>
+    <div class="container-fixed create-container">
+        <div>
+            <h1 class="page-desc">NO ACCOUNT FOUND</h1>
+        </div>
+        <div>
+            <button id="create-button" class="btn-dark btn-lg create-button">Create Account</button>
```
Comment (nwankwo-ikemefuna) [inline mini-bank/create-account.html]: This should be a link tag, it should link to the profile page.

File: mini-bank/deposits.html [hunk 3]
```diff
@@ -0,0 +1,67 @@
+<!DOCTYPE html>
```
Comment (nwankwo-ikemefuna) [inline mini-bank/deposits.html]: Rename this file to `deposit.html`, which denotes an action.

File: mini-bank/create-account.html [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Create your account</title>
+    <link rel="stylesheet" href="../helpers/css/components.css">
+    <link rel="stylesheet" href="mini-bank.css">
+</head>
+<body>
+    <div class="container-fixed create-container">
```
Comment (nwankwo-ikemefuna) [inline mini-bank/create-account.html]: `create-container` class should be wrapped inside `container-fixed`, because `container-fixed` styles can conflict with `create-container` styles.
Do this instead:
```
...
<div class="container-fixed">
    <div class="create-container">
         ...
    </div>
</div>
...
```

Also, since you're renaming this file to `landing.html`, you should change all associated namings in the file accordingly. So `create-container` class should be `landing-container`.

File: mini-bank/deposits.html [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+          <a href="user-profile.html">
+            <div class="user-nav">
+              <div class="user-icon"><i class="fas fa-circle-user"></i></div>
+              <div id="user-name" class="user-name">Username</div>
+            </div>
+           </a>
+        </div>
+
+        <div class="balance-wrapper">
```
Comment (nwankwo-ikemefuna) [inline mini-bank/deposits.html]: Move the balance wrapper above the navigation buttons.

File: mini-bank/user-profile.html [hunk 6]
```diff
@@ -0,0 +1,71 @@
+<!DOCTYPE html>
```
Comment (nwankwo-ikemefuna) [inline mini-bank/user-profile.html]: Rename this file to `account-profile.html`. Change all associated namings too.

File: mini-bank/deposits.html [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+          </div>
+          <div class="form-group">
+            <label>Pin</label>
+            <input type="password" class="form-control" id="user-pin" />
+          </div>
+          <button id="deposit-button" class="btn-success btn-lg">Submit</button>
+        </form>
+      </div>
+
+      <div class="create-account" id="create-new-account">
```
Comment (nwankwo-ikemefuna) [inline mini-bank/deposits.html]: Why do you have this button here?
The user already has an account, in the context of this project, why do they need to create another?

File: mini-bank/transactions.html [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+          <tbody id="transaction-table-body">
+            <td align="center" colspan="6">
+              <em>No transactions have been made!</em>
+            </td>
+            <!--Render feedback dynamically with JavaScript-->
+          </tbody>
+        </table>
+      </div>
+
+      <div class="create-account" id="create-new-account">
```
Comment (nwankwo-ikemefuna) [inline mini-bank/transactions.html]: This button shouldn't be here as the user already has an account. If they didn't, they won't be able to access this page.

File: mini-bank/deposits.html (hunk 7 above)
Comment (nwankwo-ikemefuna) [reply mini-bank/deposits.html]: This button shouldn't be here as the user already has an account. If they didn't, they won't be able to access this page.

File: mini-bank/withdraw.html [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+            <label>Pin</label>
+            <input type="password" class="form-control" id="user-pin" />
+          </div>
+          <button id="withdraw-button" class="btn-success btn-lg">
+            Submit
+          </button>
+        </form>
+      </div>
+
+      <div class="create-account" id="create-new-account">
```
Comment (nwankwo-ikemefuna) [inline mini-bank/withdraw.html]: Like in deposit and transactions page, this button shouldn't be here as the user already has an account. If they didn't, they won't be able to access this page.

File: mini-bank/user-profile.html [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+
+      <div class="page-desc-holder">
+        <h1 class="page-desc">My Profile</h1>
+      </div>
+
+      <div class="header-component">
+        <form id="profile-form">
+          <div class="form-group">
+            <label>Account Name</label>
+            <input type="text" class="form-control" id="user-account-name" disabled/>
```
Comment (nwankwo-ikemefuna) [inline mini-bank/user-profile.html]: Why is this field disabled? How will the user supply account name???

File: mini-bank/user-profile.html [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+          </div>
+          <div class="form-group">
+            <label>Account Number</label>
+            <input type="number" class="form-control" id="user-account-number" disabled/>
```
Comment (nwankwo-ikemefuna) [inline mini-bank/user-profile.html]: Only this field should be disabled, because the system will auto-generate a 10-digit account number for the user the first time during account creation.

File: mini-bank/user-profile.html [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+          </div>
+          <div class="form-group">
+            <label>Pin</label>
+            <input type="password" class="form-control" id="user-account-pin" disabled/>
```
Comment (nwankwo-ikemefuna) [inline mini-bank/user-profile.html]: Why is this field disabled? How will the user supply PIN???
Also, where is the confirm PIN field?

**Review by nwankwo-ikemefuna — CHANGES_REQUESTED** [review]
It's coming alive 👍 👊 . 
Just need to address the highlighted changes.

Please remove the changes not related to this feature from the PR. The changes from the projects feature should not be in this PR. Let me know if you need help with this.

File: mini-bank/mini-bank.css [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+  font-size: 25px;
+  border-radius: 10px;
+}
+.create-button:hover {
+  background: #aaa;
+  color: #000;
+  transition: 0.8s;
+}
+/*Transaction page*/
+.header-component {
```
Comment (nwankwo-ikemefuna) [inline mini-bank/mini-bank.css:39]: Would be nice to make the navigation buttons wrap nicely on smaller screens, so they don't overflow the page container.

File: mini-bank/deposits.html [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+                <a href="transactions.html"
+                ><li>Transactions</li></a
+              >
+              <a href="deposits.html" class="active"><li>Deposit</li></a>
+              <a href="withdraw.html"><li>Withdraw</li></a>
+            </ul>
+            </ul>
+          </div>
+
+          <a href="user-profile.html">
```
Comment (nwankwo-ikemefuna) [inline mini-bank/deposits.html]: Not visible in code, but you should remove the underline from the link.

File: mini-bank/user-profile.html (hunk 10 above)
Comment (john-afiukwa) [reply mini-bank/user-profile.html]: I attached an edit button there
So it can only be enabled at the click of that button

File: mini-bank/user-profile.html (hunk 12 above)
Comment (john-afiukwa) [reply mini-bank/user-profile.html]: In the users profile, there is no need for a confirm pin field.
The confirm pin field will also be displayed when the edit button is clicked

File: helpers/css/components.css [hunk 15]
```diff
@@ -32,10 +32,10 @@ button:not(:disabled):not(.disabled) {
...
   white-space: nowrap;
   vertical-align: middle;
-  -webkit-user-select: none;
-  -moz-user-select: none;
-  -ms-user-select: none;
-  user-select: none;
+  -webkit-account-select: none;
+  -moz-account-select: none;
+  -ms-account-select: none;
+  account-select: none;
```
Comment (nwankwo-ikemefuna) [inline helpers/css/components.css]: Why these changes?
Did you do a global change from _user_ to _account_?

File: helpers/js/modal.js [hunk 16]
```diff
@@ -61,13 +61,13 @@ function displayModal(modalId, headerTitle, bodyContent, footerContent = null, c
   //display the modal
   modal.style.display = "block";
 
-  // When the user clicks close, close the modal
+  // When the account clicks close, close the modal
```
Comment (nwankwo-ikemefuna) [inline helpers/js/modal.js]: This shouldn't have been updated as well

File: helpers/js/modal.js [hunk 17]
```diff
@@ -61,13 +61,13 @@ function displayModal(modalId, headerTitle, bodyContent, footerContent = null, c
...
   const modalCloseTrigger = modal.querySelector('.close');
   modalCloseTrigger.onclick = function() {
     closeModal(modal);
   }
   
-  // When the user clicks anywhere outside of the modal, close it
+  // When the account clicks anywhere outside of the modal, close it
```
Comment (nwankwo-ikemefuna) [inline helpers/js/modal.js]: This too

File: helpers/js/notes.js [hunk 18]
```diff
@@ -30,7 +30,7 @@ const storageKey = 'samplePost';
 const storageData = {
   title: 'Sample post 1',
   body: 'lorem ipsum dolor sit emeka',
-  userId: 3
+  accountId: 3
```
Comment (nwankwo-ikemefuna) [inline helpers/js/notes.js]: This should be userId.
Be careful when making global replacements so it doesn't replace unwanted things.

File: mini-bank/account-profile.js [hunk 19]
```diff
@@ -0,0 +1,65 @@
...
+    event.preventDefault();
+    if (
+      (!accountNameInput.value,
+      !accountPinInput.value,
+      !accountPinConfirmInput.value)
+    ) {
+      alert("Please fill all empty fields");
+      return;
+    } else if (accountPinInput.value !== accountPinConfirmInput.value) {
+      alert("Incorrect pin confirmation. Please confirm correct pin.");
```
Comment (nwankwo-ikemefuna) [inline mini-bank/account-profile.js]: A more appropriate error message would be "PIN and confirm PIN do not match"

File: mini-bank/account-common.js [hunk 20]
```diff
@@ -0,0 +1,40 @@
...
+
+accountNameOnHeader.textContent = accountNameDataInLocalStorage;
+
+function getCurrentBalance() {
+  const currentAccountBalance = document.getElementById(
+    "current-account-balance"
+  );
+
+  const getCurrentAccountBalance =
+    userTransactionsInLocalStorageArr[
```
Comment (nwankwo-ikemefuna) [inline mini-bank/account-common.js]: What's happening here?

File: mini-bank/account-common.js [hunk 21]
```diff
@@ -0,0 +1,40 @@
...
+      userTransactionsInLocalStorageArr.length - 1
+    ];
+  const displayActualBalance = getCurrentAccountBalance.balanceAfter;
```
Comment (nwankwo-ikemefuna) [inline mini-bank/account-common.js]: If `getCurrentAccountBalance` is null, doing .`balaceAfter` on it will error out. To check this, do:
`getCurrentAccountBalance?.balanceAfter || 0`

File: mini-bank/account-profile.html [hunk 22]
```diff
@@ -31,38 +31,50 @@
...
           </div>
           <div class="form-group">
             <label>Account Number</label>
-            <input type="number" class="form-control" id="user-account-number" disabled/>
+            <input type="number" class="form-control" id="account-number-input" disabled/>
           </div>
           <div class="form-group">
             <label>Pin</label>
-            <input type="password" class="form-control" id="user-account-pin" disabled/>
+            <input type="password" maxlength="4" class="form-control" id="account-pin-input" disabled/>
```
Comment (nwankwo-ikemefuna) [inline mini-bank/account-profile.html]: Welldone on `maxlength="4"` 👍

File: mini-bank/account-profile.js [hunk 23]
```diff
@@ -0,0 +1,65 @@
...
+  editButtonContainer.style.display = "none";
+  accountNumberInput.value = randomaccountNumber;
+
+  accountNameInput.disabled = false;
+  accountPinInput.disabled = false;
+  accountPinConfirmInput.disabled = false;
+
+  createProfileButton.addEventListener("click", (event) => {
+    event.preventDefault();
+    if (
```
Comment (nwankwo-ikemefuna) [inline mini-bank/account-profile.js:65]: You should also add check to ensure PIN is a number, using isNaN(accountPinInput). It will be false if it's a number, and true otherwise

File: mini-bank/account-profile.js [hunk 24]
```diff
@@ -0,0 +1,65 @@
...
+      accountPinConfirmInput.value = "";
+      return;
+    } else {
+      localStorage.setItem(accountNameKey, accountNameInput.value);
+      localStorage.setItem(accountNumberKey, accountNumberInput.value);
+      localStorage.setItem(accountPinKey, accountPinInput.value);
+    }
+    window.location.href = "transactions.html";
+  });
+} else {
```
Comment (nwankwo-ikemefuna) [inline mini-bank/account-profile.js:95]: If I already have an account, I should be able to also edit my account details. I can change all details except for account number. Only the account number field should be disabled.

File: mini-bank/withdrawal.js [hunk 25]
```diff
@@ -0,0 +1,55 @@
...
+
+getCurrentBalance()
+
+submitWithdrawButton.addEventListener("click", (e) => {
+  e.preventDefault();
+  if (!amountToWithdrawInput.value) {
+    alert("Please input amount to withdraw");
+    return;
+  } else if (userPinInput.value !== accountPinDataInLocalStorage) {
+    alert("Please input your correct pin");
```
Comment (nwankwo-ikemefuna) [inline mini-bank/withdrawal.js]: A more appropriate error message would be "Incorrect PIN!"

File: mini-bank/account-profile.html [hunk 26]
```diff
@@ -14,15 +14,15 @@
   </head>
   <body>
     <div class="container-fixed">
-      <section class="header-component">
+      <section id="header-component" class="header-component">
         <div class="balance-wrapper">
-          Bal: &nbsp
-          <div class="user-balance" id="user-balance">0.00</div>
+          Bal: #&nbsp
```
Comment (nwankwo-ikemefuna) [inline mini-bank/account-profile.html]: Use the naira symbol &#8358;

File: mini-bank/transactions.html [hunk 27]
```diff
@@ -56,16 +57,21 @@ <h1 class="page-desc">Transactions</h1>
               <th>Bal After</th>
             </tr>
           </thead>
-          <tbody id="transaction-table-body">
-            <td align="center" colspan="6">
-              <em>No transactions have been made!</em>
+          <tbody class="transaction-table-body" id="transaction-table-body">
```
Comment (nwankwo-ikemefuna) [inline mini-bank/transactions.html:78]: Screenshot at Sep 09 5-16-06 PM
The body cells should have normal font style, they are bold now.

**Review by nwankwo-ikemefuna — CHANGES_REQUESTED** [review]
It works as it should, so kudos 👍.
Just a few things to be addressed.

File: mini-bank/account-common.js (hunk 20 above)
Comment (john-afiukwa) [reply mini-bank/account-common.js]: That is the variable getting the transactions array from local storage 
ND then I'm getting the item on the last index to extract the balance after

Comment (nwankwo-ikemefuna) [reply mini-bank/account-common.js]: I see. You can also just do `userTransactionsInLocalStorageArr.pop()` to get the last element in the array.

**Review by nwankwo-ikemefuna — COMMENTED** [review]
Looking great now 👍
